tests/run/frozenset.pyx

```diff
@@ -0,0 +1,460 @@
+# mode: run
+
+cimport cython
+
+
+def cython_set():
+    """
+    >>> cython_set() is set
+    True
+    """
+    assert set is cython.set
+    return cython.set
+
+
+def cython_frozenset():
+    """
+    >>> cython_frozenset() is frozenset
+    True
+    """
+    assert frozenset is cython.frozenset
+    return cython.frozenset
+
+
+def cython_set_override():
+    """
+    >>> cython_set_override() is set
+    True
+    """
+    set = 1
+    return cython.set
+
+
+def cython_frozenset_override():
+    """
+    >>> cython_frozenset_override() is frozenset
+    True
+    """
+    frozenset = 1
+    return cython.frozenset
+
+
+def test_set_literal():
+    """
+    >>> type(test_set_literal()) is set
+    True
+    >>> sorted(test_set_literal())
+    ['a', 'b', 1]
+    """
+    cdef set s1 = {1,'a',1,'b','a'}
+    return s1
+
+
+def test_set_add():
+    """
+    >>> type(test_set_add()) is set
+    True
+    >>> sorted(test_set_add())
+    ['a', 1, (1, 2)]
+    """
+    cdef set s1
+    s1 = set([1, (1, 2)])
+    s1.add(1)
+    s1.add('a')
+    s1.add(1)
+    s1.add((1,2))
+    return s1
+
+
+def test_set_contains(v):
+    """
+    >>> test_set_contains(1)
+    True
+    >>> test_set_contains(2)
+    False
+    >>> test_set_contains(frozenset([1, 2, 3]))
+    True
+    >>> test_set_contains(frozenset([1, 2]))
+    False
+    >>> test_set_contains(set([1, 2, 3]))
+    True
+    >>> test_set_contains(set([1, 2]))
+    False
+    >>> try: test_set_contains([1, 2])
+    ... except TypeError: pass
+    ... else: print("NOT RAISED!")
```

Comment:
I suspect this is code that you're just moving from elsewhere, but doctest [does support testing that an exception is raised](https://docs.python.org/3/library/doctest.html#what-about-exceptions). So it's be something like:

```
>>> test_set_contains([1, 2])  # doctest: +IGNORE_EXCEPTION_DETAIL
Traceback (most recent call last):
    ...
TypeError: ...
```

(use `+IGNORE_EXCEPTION_DETAIL` or `+ELLIPSIS` as needed depending on if you care about the exact error message)

Reply:
Thanks! I've found that my PR is failing for list comprehension. I'll ask for review when I fixed it (or ask other questions here)